src/DynamoApplications/StartupUtils.cs

```diff
@@ -193,8 +193,16 @@ public static void PreloadShapeManager(ref string geometryFactoryPath, ref strin
         public static DynamoModel MakeCLIModel(CommandLineArguments cmdLineArgs)
         {
             var asmPath = String.IsNullOrEmpty(cmdLineArgs.ASMPath) ? string.Empty : cmdLineArgs.ASMPath;
-            IPathResolver pathResolver = CreatePathResolver(false, string.Empty, string.Empty, string.Empty);
+
+            IPathResolver pathResolver = CreatePathResolver(true, string.Empty, cmdLineArgs.UserDataFolder, cmdLineArgs.CommonDataFolder);
             PathManager.Instance.AssignHostPathAndIPathResolver(string.Empty, pathResolver);
+
+            if (!File.Exists(PathManager.Instance.PreferenceFilePath))
```

Comment:
will this traverse the newly set user data folder from the command line args and check for a prefs file there?

Reply:
Yes. If `userData` parameter is, e.g. `%appData%\Dynamo\Dynamo Revit`, `PathManager.Instance.PreferenceFilePath` is `%appData%\Dynamo\Dynamo Revit\3.4\DynamoSettings.xml`.